File: MagicalSticks.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    // variable declaration and input
    int t;
    cin >> t;
    vector<int> sol;
    while (t--) {
        int n;
        cin >> n;
        // i just guessed and it was correct lol
        if (n==1)
            sol.push_back(1);
        else if (n%2!=0)
            sol.push_back(n/2 + 1);
        else if (n%2==0)
            sol.push_back(n/2);
    }
    for (auto &x:sol)
        cout << x << endl;

    return 0;
}

File: Python/MagicalSticks.py
def main():
    t = int(input())
    sol = []
    while t:
        n = int(input())
        if n == 1:
            sol.append(1)
        elif n % 2 == 0:
            sol.append(n/2+1)
        else:
            sol.append(n/2)
        t -= 1

    for i in sol:
        print(i)


if __name__ == '__main__':
    main()
